# global robot configuration and details

robot_config:

    joint_names:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7
    arm_id: panda
        # Activate rate limiter? [true|false]
    joint_limit_warning_threshold: 0.1 # [rad]
    rate_limiting: true
    # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
    cutoff_frequency: 100
    # Internal controller for motion generators [joint_impedance|cartesian_impedance]
    internal_controller: cartesian_impedance

    neutral_pose:
        panda_joint1: -0.017792060227770554 
        panda_joint2: -0.7601235411041661  
        panda_joint3: 0.019782607023391807 
        panda_joint4: -2.342050140544315 
        panda_joint5: 0.029840531355804868 
        panda_joint6: 1.5411935298621688 
        panda_joint7: 0.7534486589746342 

    joint_config:
        joint_acceleration_limit:
            panda_joint1: 15.0 # rad / sec^2
            panda_joint2: 7.5  # rad / sec^2
            panda_joint3: 10.0 # rad / sec^2
            panda_joint4: 12.5 # rad / sec^2
            panda_joint5: 15.0 # rad / sec^2
            panda_joint6: 20.0 # rad / sec^2
            panda_joint7: 20.0 # rad / sec^2
            # panda_finger_joint1: 12.0 # rad / sec^2
            # panda_finger_joint2: 12.0 # rad / sec^2

        joint_velocity_limit:
            panda_joint1: 2.1750 # rad / sec
            panda_joint2: 2.1750 # rad / sec
            panda_joint3: 2.1750 # rad / sec
            panda_joint4: 2.1750 # rad / sec
            panda_joint5: 2.6100 # rad / sec
            panda_joint6: 2.6100 # rad / sec
            panda_joint7: 2.6100 # rad / sec
            # panda_finger_joint1: 2.0 # rad / sec
            # panda_finger_joint2: 2.0 # rad / sec

        joint_position_limit:
            lower: 
                panda_joint1: -2.8973 # rad
                panda_joint2: -1.7628 # rad
                panda_joint3: -2.8973 # rad
                panda_joint4: -3.0718 # rad
                panda_joint5: -2.8973 # rad
                panda_joint6: -0.0175 # rad
                panda_joint7: -2.8973 # rad
            upper:
                panda_joint1: 2.8973 # rad
                panda_joint2: 1.7628 # rad
                panda_joint3: 2.8973 # rad
                panda_joint4: -0.0698 # rad
                panda_joint5: 2.8973 # rad
                panda_joint6: 3.7525 # rad
                panda_joint7: 2.8973 # rad

        joint_effort_limit:
            panda_joint1: 87 # Nm
            panda_joint2: 87 # Nm
            panda_joint3: 87 # Nm
            panda_joint4: 87 # Nm
            panda_joint5: 12 # Nm
            panda_joint6: 12 # Nm
            panda_joint7: 12 # Nm      

    realtime_config: enforce
    # Configure the initial defaults for the collision behavior reflexes.

    # collision_config:
    #   lower_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    #   upper_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]

    #   lower_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    #   upper_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]

    #   lower_force_thresholds_acceleration: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_acceleration: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
      
    #   lower_force_thresholds_nominal: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_nominal: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]     

    #   robot.setJointImpedance({{3000, 3000, 3000, 2500, 2500, 2000, 2000}});
    #   robot.setCartesianImpedance({{3000, 3000, 3000, 300, 300, 300}});

    # collision_config:
    #   lower_torque_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    #   upper_torque_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]

    #   lower_torque_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    #   upper_torque_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]

    #   lower_force_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
      
    #   lower_force_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]

    # collision_config:
    #   lower_torque_thresholds_acceleration: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [Nm]
    #   upper_torque_thresholds_acceleration: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [Nm]

    #   lower_torque_thresholds_nominal: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [Nm]
    #   upper_torque_thresholds_nominal: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [Nm]

    #   lower_force_thresholds_acceleration: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_acceleration: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [N, N, N, Nm, Nm, Nm]
      
    #   lower_force_thresholds_nominal: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [N, N, N, Nm, Nm, Nm]
    #   upper_force_thresholds_nominal: [200.0, 200.0, 200.0, 200.0, 200.0, 200.0, 200.0]  # [N, N, N, Nm, Nm, Nm]

    collision_config:
      lower_torque_thresholds_acceleration: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [Nm]
      upper_torque_thresholds_acceleration: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [Nm]

      lower_torque_thresholds_nominal: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [Nm]
      upper_torque_thresholds_nominal: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [Nm]

      lower_force_thresholds_acceleration: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [N, N, N, Nm, Nm, Nm]
      upper_force_thresholds_acceleration: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [N, N, N, Nm, Nm, Nm]
      
      lower_force_thresholds_nominal: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [N, N, N, Nm, Nm, Nm]
      upper_force_thresholds_nominal: [400.0, 400.0, 400.0, 400.0, 400.0, 400.0, 400.0]  # [N, N, N, Nm, Nm, Nm]



gripper_config:
  joint_names:
    - panda_finger_joint1
    - panda_finger_joint2
  default_speed: 0.1  # [m/s]
  default_grasp_epsilon:
    inner: 0.005 # [m]
    outer: 0.005 # [m]

controllers_config:
    position_controller: "franka_ros_interface/position_joint_position_controller"
    torque_controller: "franka_ros_interface/effort_joint_torque_controller"
    impedance_controller: "franka_ros_interface/effort_joint_impedance_controller"
    velocity_controller: "franka_ros_interface/velocity_joint_velocity_controller"
    trajectory_controller: "position_joint_trajectory_controller"
    default_controller: "position_joint_trajectory_controller" # for safety, always set a position controller as default
    command_timeout: 0.2 # timeout to wait for consecutive torque control commands (or velocity) when using torque (or velocity) control. If timeout is violated, the controller interface will automatically switch to default controller for safety